src/windows/mod.rs

```diff
@@ -56,23 +58,29 @@ impl SystemReaderInterface for SystemReader {
     fn get_wmi_drives_info(&self) -> Option<String> {
         let output = Command::new("wmic")
             .args(&["logicaldisk", "get", "DeviceID,FileSystem,Size,FreeSpace,DriveType",
-                    "/format:list"]).output().ok()?;
+                "/format:list"]).output().ok()?;
         String::from_utf8(output.stdout).ok()
     }
 
     fn get_wmi_nicconfig(&self) -> Option<String> {
         let output = Command::new("wmic")
             .args(&["nicconfig", "get",
-                    "IPEnabled,InterfaceIndex,Description,DefaultIPGateway,IPAddress,IPSubnet,DHCPEnabled",
-                    "/format:list"]).output().ok()?;
+                "IPEnabled,InterfaceIndex,Description,DefaultIPGateway,IPAddress,IPSubnet,DHCPEnabled",
+                "/format:list"]).output().ok()?;
         String::from_utf8(output.stdout).ok()
     }
-    
+
     fn get_wmi_nicconfig_details(&self) -> Option<String> {
         let output = Command::new("wmic")
             .args(&["nicconfig", "get", "IPEnabled,InterfaceIndex,MACAddress,MTU", "/format:list"]).output().ok()?;
         String::from_utf8(output.stdout).ok()
     }
+
+    fn get_hosts_file(&self) -> Option<String> {
+        let mut s = String::new();
+        File::open("c:\\windows\\system32\\drivers\\etc\\hosts").ok()?.read_to_string(&mut s).ok()?;
```

Comment:
C:\Windows\System32 should be dynamically detected from either an environment variable like %SYSTEMROOT%\System32

Reply:
std:env::var could be used to get the value of SYSTEMROOT.
https://doc.rust-lang.org/std/env/fn.var.html